src/compiler/mir/transform.rs

```diff
@@ -7,22 +7,336 @@
+            }
+            Expression::Null(_) => todo!(),
+            Expression::U8(_, _) => todo!(),
+            Expression::U16(_, _) => todo!(),
+            Expression::U32(_, _) => todo!(),
+            Expression::U64(_, _) => todo!(),
+            Expression::I8(_, _) => todo!(),
+            Expression::I16(_, _) => todo!(),
+            Expression::I32(_, _) => todo!(),
+            Expression::F64(_, _) => todo!(),
+            Expression::Boolean(_, b) => self.gen.const_bool(*b),
+            Expression::StringLiteral(_, _) => todo!(),
+            Expression::ArrayExpression(_, _, _) => todo!(),
+            Expression::ArrayAt {
+                context,
+                array,
+                index,
+            } => todo!(),
+            Expression::SizeOf(_, _) => todo!(),
+            Expression::CustomType(_, _) => todo!(),
+            Expression::Identifier(_, id) => {
+                // Look up Var ID using the Identifier String ID
+                let vid = self.gen.find_var(*id).unwrap();
+
+                // Return a LValue::Var(VarId) as the result of this expression
+                Operand::LValue(LValue::Var(vid))
+            }
+            Expression::Path(_, _) => todo!(),
+            Expression::MemberAccess(_, _, _) => todo!(),
+            Expression::IdentifierDeclare(_, _, _) => todo!(),
+            Expression::RoutineCall(_, _, _, _) => todo!(),
+            Expression::StructExpression(_, _, _) => todo!(),
+            Expression::If {
+                context,
+                cond,
+                if_arm,
+                else_arm,
+            } => self.if_expr(cond, if_arm, else_arm),
+            Expression::While {
+                context,
+                cond,
+                body,
+            } => todo!(),
+            Expression::ExpressionBlock(_, block, expr) => {
+                for stm in block {
+                    self.statement(stm);
+                }
+                if let Some(expr) = expr {
+                    self.expression(expr)
+                } else {
+                    todo!()
+                }
+            },
+            Expression::TypeCast(_, _, _) => todo!(),
+            Expression::UnaryOp(_, _, _) => todo!(),
+            Expression::Yield(_, _) => todo!(),
+        }
+    }
+
+    fn if_expr(
+        &mut self,
+        cond: &Expression<SemanticContext>,
+        then_block: &Expression<SemanticContext>,
+        else_block: &Option<Box<Expression<SemanticContext>>>,
+    ) -> Operand {
+        let then_bb = self.gen.new_bb();
+        let else_bb = self.gen.new_bb();
+        let merge_bb = self.gen.new_bb();
+        let cond_val = self.expression(cond);
+        self.gen.term_if(cond_val, then_bb, else_bb);
+
+        // if the if expression has a type other than unit, then create a temporary
+        // variable to store the resolved value.
+        let temp = self.gen.temp(then_block.get_type());
```

Comment:
If the type of `then` is Unit or if else_bb is `None`, then there is no need to create storage for this expression